src/snmalloc/global/memcpy.h

```diff
@@ -179,6 +189,60 @@ namespace snmalloc
     }
   };
 
+  /**
+   * StrictProvenance architectures are prickly about their pointers.  In
+   * particular, they may not permit misaligned loads and stores of
+   * pointer-sized data, even if they can have non-pointers in their
+   * pointer registers.  On the other hand, pointers might be hiding anywhere
+   * they are architecturally permitted!
+   */
+  struct GenericStrictProvenance
+  {
+    static SNMALLOC_FAST_PATH_INLINE void
+    copy(void* dst, const void* src, size_t len)
+    {
+      static_assert(bits::is_pow2(sizeof(void*)));
+
+      // If this is a small size, use a jump table
+      if (len < sizeof(void*))
+      {
+        small_copies<sizeof(void*), sizeof(long)>(dst, src, len);
```

Comment:
```suggestion
        small_copies<sizeof(void*)-1, sizeof(long)>(dst, src, len);
```
I missed the `<` above and so was confused about why this works.